src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,112 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import java.util.List;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+@TestInstance(PER_CLASS)
+public class RegistrationServiceTest {
+    private RegistrationServiceImpl registrationService;
+    private StorageDaoImpl dao;
+    private List<User> validUsers;
+
+    @BeforeAll
+    public void setup() {
+        registrationService = new RegistrationServiceImpl();
+        dao = new StorageDaoImpl();
+        validUsers = List.of(new User("James", "qwerty",20),
+                        new User("Anna", "password", 30),
+                        new User("Alice", "123456", 40));
+    }
+
+    @AfterAll
+    public void close(){
+    }
+
+    @Test
+    public void register_nullUser_throwsNullPointer() {
```

Comment:
```suggestion
    public void register_nullUser_notOk() {
```

same for other test names